Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -3231,6 +3235,9 @@ private void getPlayableFromGraveyardCard(Game game, Card card, Abilities<Abilit
                         AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, card.getSpellAbility(), getId(), game)) {
                     playable.add(ability);
                 }
+                if (setControllerId) {
+                    ability.setControllerId(card.getOwnerId());
+                }
```

Comment:
That's was workaround to temporary change ability's controller and must be return to OLD value after all checks (see another comment about missing `canActive` code).

From old code:
![shot_191211_004919](https://user-images.githubusercontent.com/8344157/70567966-615e8580-1bb0-11ea-9d0a-ac5a2053124f.png)

I'm also recommends save and revert to old controllerId, not ownerId (current version with ownerId can broke game data potentially).

Reply:
Yep, it's currently functionally the same as the old code but I can change it to save + restore instead of putting back to owner.

Reply:
Yes, restore controllerId as old value, not ownerId.